components/Publication/index.js

```diff
@@ -0,0 +1,254 @@
+      id
+      isArchived
+      meta {
+        publishDate
+      }
+      latestPublications {
+        date
+        name
+        live
+        prepublication
+        scheduledAt
+      }
+      commit(id: $commitId) {
+        id
+        message
+        date
+        author {
+          email
+          name
+        }
+        document {
+          id
+          repoId
+          content
+          meta {
+            slug
+            emailSubject
+            template
+            title
+            description
+            image
+            facebookDescription
+            facebookImage
+            facebookTitle
+            twitterDescription
+            twitterImage
+            twitterTitle
+            format {
+              id
+              repoId
+              meta {
+                path
+                title
+                color
+                kind
+              }
+            }
+            section {
+              id
+              repoId
+              meta {
+                path
+                title
+                color
+                kind
+              }
+            }
+            series {
+              title
+              logo
+              logoDark
+              episodes {
+                title
+                label
+                document {
+                  id
+                  repoId
+                  meta {
+                    path
+                  }
+                }
+              }
+            }
+          }
+          subscribedBy(
+            includeParents: true
+            onlyEligibles: true
+            uniqueUsers: true
+          ) {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+
+const styles = {
+  container: css({
+    display: 'flex'
+  }),
+  formContainer: css({
+    position: 'fixed',
+    height: '100%',
+    right: 0,
+    width: '500px',
+    padding: '16px 24px 100px 24px',
+    backgroundColor: 'white',
+    overflow: 'scroll',
+    overscrollBehavior: 'contain'
+  }),
+  column: css({
+    padding: '1em'
+  }),
+  darkmodeButton: css({
+    position: 'absolute',
+    margin: '-26px 0 0 32px',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center'
+  })
+}
+
+const PublishForm = ({ t, data, publish }) => {
+  const { loading, error, repo } = data
+  const [previewScreenSize, setPreviewScreenSize] = useState('phone')
+  const [previewDarkmode, setPreviewDarkmode] = useState(false)
+
+  return (
+    <>
+      <Loader
+        loading={loading}
+        error={error}
+        render={() => {
+          const { isArchived, commit } = repo
+
+          if (isArchived) {
+            return <RepoArchivedBanner />
+          }
+
+          return (
+            <>
+              <div {...styles.formContainer}>
+                <PublicationForm
+                  t={t}
+                  repo={repo}
+                  commit={commit}
+                  publish={publish}
```

Comment:
I don't quite understand the logic of pulling the publish mutation out of the `PublicationForm`, only to pass it as prop to said form. What's the reason behind this change?

Reply:
I didn't change anything here. I just split up a file that was too big into an index.js and PublicationFrom.js The contents remain untouched.